packages/web/components/bridge/immersive/crypto-fiat-input.tsx

```diff
@@ -0,0 +1,226 @@
+                }),
+              }}
+              onBlur={() => setIsInputFocused(false)}
+              onFocus={() => setIsInputFocused(true)}
+              currentValue={fiatCurrentValue}
+              onInput={onInput("fiat")}
+              isAutosize
+            />
+          ) : (
+            <InputBox
+              // when the font size changes, we need to prompt the autosize input to re-mount
+              // see: https://github.com/JedWatson/react-input-autosize?tab=readme-ov-file#changing-the-styles-at-runtime
+              key={cryptoInputFontSize}
+              inputRef={inputRef}
+              className={classNames(
+                "border-none bg-transparent font-bold",
+                cryptoInputFontSize
+              )}
+              classes={{
+                label: "!block",
+                input: classNames("!p-0", {
+                  "text-rust-300": isInsufficientBal || isInsufficientFee,
+                }),
+                trailingSymbol: classNames(
+                  "ml-1 align-middle text-osmoverse-500",
+                  {
+                    "text-rust-300": isInsufficientBal || isInsufficientFee,
+                  }
+                ),
+              }}
+              onBlur={() => setIsInputFocused(false)}
+              onFocus={() => setIsInputFocused(true)}
+              currentValue={cryptoInputRaw}
+              onInput={onInput("crypto")}
+              trailingSymbol={cryptoAmountPretty.denom}
+              isAutosize
+            />
+          )}
+        </div>
+        <button
+          onClick={() => {
+            // TODO: Subtract fees from the amount
+            onInput("crypto")(
+              trimPlaceholderZeros(asset.amount.toDec().toString())
+            );
+          }}
+          className="body2 w-14 shrink-0 transform rounded-5xl border border-osmoverse-700 py-2 px-3 text-wosmongton-200 transition duration-200 hover:border-osmoverse-850 hover:bg-osmoverse-850 hover:text-white-full disabled:opacity-80"
+        >
+          {t("transfer.max")}
+        </button>
+      </div>
+
+      <button
+        className="body1 flex items-center gap-2 text-center text-wosmongton-200"
+        onClick={() => {
+          setInputUnit(currentUnit === "fiat" ? "crypto" : "fiat");
+        }}
+      >
+        <span>
+          {currentUnit === "fiat" ? (
+            <>
+              {trimPlaceholderZeros(
+                cryptoAmountPretty?.toDec().toString(2) ?? "0"
+              )}{" "}
+              {cryptoAmountPretty.denom}
+            </>
+          ) : (
+            fiatAmountPretty.maxDecimals(2).toString()
+          )}
+        </span>
+        <span>
+          <Icon
+            id="switch"
+            className="text-wosmongton-200"
+            width={16}
+            height={16}
+          />
+        </span>
+      </button>
+    </div>
+  );
+};
+
+function calcTextSizeClass(numChars: number): string {
```

Comment:
something like this might be more elegant:

```
const calcTextSizeClass = (numChars: number): string => {
  const sizeMapping: { [key: number]: string } = {
    8: "text-4xl",
    10: "text-3xl",
    12: "text-2xl",
    18: "text-xl",
    24: "text-lg",
  };

  for (const [key, value] of Object.entries(sizeMapping)) {
    if (numChars <= Number(key)) {
      return value;
    }
  }

  return "text-md";
};
```